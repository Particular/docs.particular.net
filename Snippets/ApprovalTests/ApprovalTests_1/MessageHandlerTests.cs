using System.Threading.Tasks;
using NServiceBus.ApprovalTests;
using NServiceBus.Testing;
using NUnit.Framework;

[TestFixture]
public class MessageHandlerTests
{
    #region HandlerTest
    [Test]
    public async Task VerifyHandlerResult()
    {
        var handler = new MyHandler();
        var context = new TestableMessageHandlerContext();

        await handler.Handle(new MyRequest(), context)
            .ConfigureAwait(false);

        TestContextVerifier.Verify(context);
    }
    #endregion
}