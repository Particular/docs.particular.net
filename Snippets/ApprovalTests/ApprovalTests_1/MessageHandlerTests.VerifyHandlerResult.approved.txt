{
  "RepliedMessages": [
    {
      "MyReplyMessage": {
        "Property": "Value"
      }
    }
  ],
  "ForwardedMessages": [
    "newDestination"
  ],
  "MessageId": "Guid 1",
  "ReplyToAddress": "reply address",
  "SentMessages": [
    {
      "MySendMessage": {
        "Property": "Value"
      },
      "Options": {
        "DeliveryDelay": "12:00:00"
      }
    }
  ],
  "PublishedMessages": [
    {
      "MyPublishMessage": {
        "Property": "Value"
      }
    }
  ],
  "Extensions": {}
}